clients/pkg/promtail/targets/azureeventhubs/parser.go

```diff
@@ -53,6 +54,36 @@ func (l azureMonitorResourceLog) validate() error {
 	return nil
 }
 
+func (l azureMonitorResourceLog) isTimeOrTimeStampFieldSet() bool {
+	return len(l.Time) != 0 || len(l.TimeStamp) != 0
+}
+
+// getTime returns time from `time` or `timeStamp` field. If both fields are set, `time` is used. If both fields are empty, error is returned.
+func (l azureMonitorResourceLog) getTime() (time.Time, error) {
+	var t time.Time
+	var err error
+
+	if len(l.Time) == 0 && len(l.TimeStamp) == 0 {
+		return t, errors.New("time and timeStamp fields are empty")
+	}
+
+	if len(l.TimeStamp) != 0 {
+		t, err = time.Parse(time.RFC3339, l.TimeStamp)
+		if err != nil {
+			return t, err
+		}
+	}
+
+	if len(l.Time) != 0 {
+		t, err = time.Parse(time.RFC3339, l.Time)
+		if err != nil {
+			return t, err
+		}
+	}
+
+	return t.UTC(), nil
+}
```

Comment:
I think this reads a little easier (and quicker) to return after each possibility, and evaluate the properties in the same order we prioritize them. WDYT?

```suggestion
// getTime returns time from `time` or `timeStamp` field. If both fields are set, `time` is used. If both fields are empty, error is returned.
func (l azureMonitorResourceLog) getTime() (time.Time, error) {
	if len(l.Time) == 0 && len(l.TimeStamp) == 0 {
		return t, errors.New("time and timeStamp fields are empty")
	}
	
	if len(l.Time) != 0 {
		t, err := time.Parse(time.RFC3339, l.Time)
		if err != nil {
			return t, err
		}
		
		return t.UTC(), nil
	}

        t, err := time.Parse(time.RFC3339, l.TimeStamp)
        if err != nil {
	        return t, err
        }

	return t.UTC(), nil
}
```

Reply:
@trevorwhitney, thank you! Looks good to me 👍 

I will use this suggestion and add `t` initialization to the block because `t` is not defined in this scope.

```
if len(l.Time) == 0 && len(l.TimeStamp) == 0 {
		  return t, errors.New("time and timeStamp fields are empty")
}
```

In addition, I will add one more test similar to `Test_parseMessage_message_without_time_with_time_stamp` to explicitly test behavior when none of the fields is present in the input.